packages/hardhat-core/src/internal/solidity/compiler/downloader.ts

```diff
@@ -12,6 +13,19 @@ export interface CompilerBuild {
   longVersion: string;
   keccak256: string;
   urls: string[];
+  platform: CompilerPlatform;
+}
+
+export enum CompilerPlatform {
+  LINUX = "linux-amd64",
+  WINDOWS = "windows-amd64",
+  MACOS = "macosx-amd64",
+  DEFAULT = "bin",
```

Comment:
I'd use `WASM = "wasm"` instead of this. It should be more performant in older solc versions.

Reply:
Done. Worked fine in linux. Should I re-test this on windows?

Reply:
Don't we have some automated tests for this? I'm pretty sure some of the plugin tests log "downloading the compiler"

Reply:
Oh, those would use the native version, right?

Reply:
Yes, but maybe we can use the `forceSolcJs` to test both things. I will do that.